DOCUMENTATION.md

```diff
@@ -1,810 +1,807 @@
-
-|`error.code`|`error.message`|
-|:---|:---|
-|`404`|`[{`<br>`"error": "NoResultFound",`<br>`"message": "No result found"`<br>`}]`|
-|`404`|`[{`<br>`"error": "ValidationError",`<br>`"message": "is is not a valid UUID"`<br>`}]`|
+#### version (required)
 
-</details>
+The version number of the template.
 
+### Response
 
-#### Arguments
+If the request to the client is successful, the client will return a `Template`:
 
-<details>
-<summary>
-Click here to expand for more information.
-</summary>
+_NEED EXAMPLE_
 
-#### `id`
-The template id is visible on the template page in the application.
+### Error codes
 
-#### `version`
-A history of the template is kept. There is a link to `See previous versions` on the template page in the application.
-
-</details>
+If the request is not successful, the client will return a `Notifications::Client::RequestError` containing the relevant error code:
 
+|error.code|error.message|How to fix|
+|:---|:---|:---|
+|`404`|`[{`<br>`"error": "NoResultFound",`<br>`"message": "No result found"`<br>`}]`|
+|`400`|`[{`<br>`"error": "ValidationError",`<br>`"message": "id is not a valid UUID"`<br>`}]`|Check the notification ID|
+|`403`|`[{`<br>`"error": "AuthError",`<br>`"message": "Error: Your system clock must be accurate to within 30 seconds"`<br>`}]`|Check your system clock|
+|`403`|`[{`<br>`"error": "AuthError",`<br>`"message": "Invalid token: signature, api token not found"`<br>`}]`|Use the correct API key. Refer to [API keys](/#api-keys) for more information|
+|`404`|`[{`<br>`"error": "NoResultFound",`<br>`"message": "No Result Found"`<br>`}]`|Check your [template ID](/#get-a-template-by-id-and-version-arguments-template-id-required) and [version](/#version)|
 
 ## Get all templates
 
-#### Method
+### Method
 
-This will return the latest version of each template for your service.
-
-<details>
-<summary>
-Click here to expand for more information.
-</summary>
+This will return the latest version of all templates inside a collection named `templates`.
 
 ```ruby
-args = {
-  'template_type' => 'sms'
+args: {
+  'type' => 'sms'
 }
-templates = client.get_all_templates(args)
+templates: client.get_all_templates(args)
 ```
 
+### Arguments
 
-</details>
+#### type (optional)
 
+If you do not use `type`, the client returns all templates. Otherwise you can filter by:
 
-#### Response
+- `email`
+- `sms`
+- `letter`
 
-<details>
-<summary>
-Click here to expand for more information.
-</summary>
+### Response
+
+If the request to the client is successful, the client will return a `Notifications::Client::TemplateCollection`:
 
 ```ruby
-    TemplateCollection templates;
+TemplateCollection "TemplateCollectionName";
 ```
-If the response is successful, a TemplateCollection is returned.
 
-If no templates exist for a template type or there no templates for a service, the templates list will be empty.
+You must then run `TemplateCollectionName.collection` to return an array of the required templates.
```

Comment:
Same comment applies here as for the `Notifications::Client::NotificationsCollection` earlier - it's better to say you can call the `templates.collection` method to return the array.